projects/kit/components/input-password/input-password.component.ts

```diff
@@ -54,8 +73,22 @@ export class TuiInputPasswordComponent
         private readonly textfieldSize: TuiTextfieldSizeDirective,
         @Inject(TUI_PASSWORD_TEXTS)
         readonly passwordTexts$: Observable<[string, string]>,
+        @Inject(TUI_INPUT_PASSWORD_OPTIONS)
+        public readonly options: InputPasswordOptions,
+        @Inject(TUI_HINT_WATCHED_CONTROLLER)
+        readonly hintController: TuiHintControllerDirective,
+        @Inject(TuiDestroyService) destroy$: Observable<unknown>,
+        @Inject(TUI_MODE) mode$: Observable<TuiBrightness | null>,
     ) {
         super(control, changeDetectorRef);
+
+        mode$.pipe(takeUntil(destroy$)).subscribe(mode => {
```

Comment:
Just came to my mind that we can get rid of the subscription here in favor of `async` pipe:

```ts
readonly computedMode$ = combineLatest(
  this.mode$,
  this.hintController.change$.pipe(map(() => this.hintController.mode)),
).pipe(
  map(([mode, controller]) => controller || mode),
  startWith(null)
)
```
Just need to check that both emit initially.

Reply:
hintController emits changes before component constructor executing. So combineLatest needs startWith operator inside hintController observable. I've done it